main/internal/implementations/terraform_workspace/search_repo_workspaces_test.go

```diff
@@ -0,0 +1,128 @@
+package terraformWorkspace
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTerraformCloud_cleanDirectoryName(t *testing.T) {
+	type args struct {
+		directory string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "happy path",
+			args: args{
+				directory: "google-backend-api-dev",
+			},
+			want: "google-backend-api-dev",
+		},
+		{
+			name: "happy path with trailing slash",
+			args: args{
+				directory: "google-backend-api-dev/",
+			},
+			want: "google-backend-api-dev",
+		},
+		{
+			name: "happy path with trailing slash and whitespace",
+			args: args{
+				directory: "google-backend-api-dev/ ",
+			},
+			want: "google-backend-api-dev",
+		},
+		{
+			name: "happy path with trailing at beginning and end",
+			args: args{
+				directory: "/google-backend-api-dev/",
+			},
+			want: "google-backend-api-dev",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.Equalf(t, tt.want, cleanDirectoryName(tt.args.directory), "cleanDirectoryName(%v)", tt.args.directory)
+		})
+	}
+}
+
+func TestExtractBackendDetails(t *testing.T) {
+	// Given
+	ctx := context.Background()
+
+	filePath := "test.tfstate"
```

Comment:
`hclsimple` in the docs specifies that this needs to be either `.hcl` or `.json`: https://github.com/hashicorp/hcl/blob/v2.17.0/hclsimple/hclsimple.go#L68